src/auth/utils.ts

```diff
@@ -558,7 +558,10 @@ export class AuthNode implements TreeNode<Auth> {
 
     public getTreeItem() {
         // Calling this here is robust but `TreeShim` must be instantiated lazily to stop side-effects
-        this.resource.tryAutoConnect()
+        // TODO: this is a race!
+        this.resource.tryAutoConnect().catch(e => {
```

Comment:
This looks like a bug. The code below seems to depend on the outcome of `tryAutoConnect`.